include/core/JamomaSample.h

```diff
@@ -244,7 +254,92 @@ namespace Jamoma {
 			for (int i=0; i<frameCount; ++i)
 				values[i] = vector[i];
 		}
-		
+        
+        
+        /** Duplicate values at the beginning and end to provide padding for interpolation.
+            @param  paddingAmount   Number of frames to add to both beginning & end of SampleBundle. Total number of frames added to SampleBundle will be twice this amount.
+         */
+        void applySamplePadding(int paddingAmount)
+        {
+            
+            // ensure that we are adding at least one sample, AND not more than half SampleBundle frames
+            assert(paddingAmount > 0 && paddingAmount < mFrameCount/2);
+            
+            // make a copy
+            SampleBundleData newPaddedChannels = mChannels;
+            
+            for (auto& channel : newPaddedChannels) {
+                
+                // set up temporary vectors for padding samples
+                SampleVector firstSamples(paddingAmount,0.0);
+                SampleVector lastSamples(paddingAmount,0.0);
+                
```

Comment:
I don't see a way around steps 1 and 2. We want to make sure that we aren't running `applySamplePadding()` while another operation is in progress. That could cause some major problems.

I could see some ways to consolidate lines 273-281, which would be steps 3-7 in your list. I'll make this a new issue.